ubuntu_image/tests/test_parser.py

```diff
@@ -347,12 +318,11 @@ def test_mbr_structure_conflicting_filesystem(self):
   first-image:
     bootloader: u-boot
     structure:
-        - type: mbr
-          size: 400M
-          filesystem: ext4
+        - type: 00000000-0000-0000-0000-0000deadbeef
 """)
-        self.assertEqual(str(cm.exception),
-                         'mbr type must not specify a file system')
+        self.assertEqual(
+            str(cm.exception),
+            'Invalid gadget.yaml @ volumes:first-image:structure:0:size')
```

Comment:
This seems wrong, since the test is specifically checking that `type: mbr` is incompatible with `filesystem:`.  That should probably just change to `role: mbr` and continue to verify the incompatibility with `filesystem:`.

Although, hmm, it looks like you might have added such a test below.  In which case, because of the legacy behavior, just keep this one unchanged.